pkg/identityserver/rights.go

```diff
@@ -213,3 +214,114 @@ func (is *IdentityServer) UserRights(ctx context.Context, userIDs *ttnpb.UserIde
 	}
 	return entity.Union(universal), nil
 }
+
+var (
+	errInsufficientRights = errors.DefinePermissionDenied(
+		"insufficient_rights",
+		"insufficient rights",
+	)
+	errSomeGatewaysNotFound = errors.DefineNotFound(
+		"some_gateways_not_found",
+		"some gateways not found",
+	)
+)
+
+func (is *IdentityServer) assertGatewayRights( // nolint:gocyclo
+	ctx context.Context,
+	gtwIDs []*ttnpb.GatewayIdentifiers,
+	requiredGatewayRights *ttnpb.Rights,
+) error {
+	authInfo, err := is.authInfo(ctx)
+	if err != nil {
+		return err
+	}
+
+	// If the caller is not an organization or user, there's nothing more to do.
+	ouID := authInfo.GetOrganizationOrUserIdentifiers()
+	if ouID == nil {
+		return errInsufficientRights.New()
```

Comment:
Probably worth having a different error informing that the request was made by someone who isn't authenticated. Because the only case where a requests is made to batch like this would be by interacting with the raw API (not via console or CLI) without providing the necessary data on the request's header.